js/state/siteUtil.js

```diff
@@ -107,6 +127,28 @@ module.exports.isSiteBookmarked = function (sites, siteDetail) {
   return false
 }
 
+/**
+ * Checks if a location is bookmarked.
+ *
+ * @param state The application state Immutable map
+ * @param {string} location
+ * @return {boolean}
+ */
+module.exports.isLocationBookmarked = function (state, location) {
+  const sites = state.get('sites')
+  const siteKeys = siteCache.getLocationSiteKeys(state, location)
+  if (!siteKeys || siteKeys.length === 0) {
+    return false
+  }
+  return siteKeys.some(key => {
+    const site = sites.get(key)
+    if (!site) {
+      return false
+    }
```

Comment:
some will still iterate over everything, I think it would be best to just make the cache related to bookmarks, and if the key is in there then return in O(1) if it is bookmarked.